modules/javafx.media/src/main/native/jfxmedia/jni/JniUtils.cpp

```diff
@@ -42,21 +42,22 @@ void ThrowJavaException(JNIEnv *env, const char* type, const char* message)
     jclass klass = NULL;
     if (type) {
         klass = env->FindClass(type);
-        if (!klass) {
-            // might have caused an exception
-            if (env->ExceptionOccurred()) {
-                env->ExceptionClear();
-            }
+        // might have caused an exception
+        if (env->ExceptionCheck()) {
+            env->ExceptionClear();
         }
     }
     if (!klass) {
         klass = env->FindClass("java/lang/Exception");
+        if (env->ExceptionCheck() || klass != NULL) {
+            env->ExceptionClear();
+            return; // This shouldn't happen...
+        }
         if (!klass) {
-            if (env->ExceptionOccurred()) {
+            if (env->ExceptionCheck()) {
                 env->ExceptionClear();
             }
-            // This shouldn't happen...
-            return;
```

Comment:
Don't you still need this `return`?

Reply:
I moved it under `if (env->ExceptionCheck() || klass != NULL)`, but this check is incorrect and it should be `if (env->ExceptionCheck() || klass == NULL)`. I fixed it.